skglm/datafits/group.py

```diff
@@ -63,6 +78,24 @@ def gradient_g(self, X, y, w, Xw, g):
 
         return grad_g
 
+    def gradient_g_sparse(self, X_data, X_indptr, X_indices, y, w, Xw, g):
+        grp_ptr, grp_indices = self.grp_ptr, self.grp_indices
+        grp_g_indices = grp_indices[grp_ptr[g]: grp_ptr[g+1]]
+
+        grad_g = np.zeros(len(grp_g_indices))
+        for idx, j in enumerate(grp_g_indices):
+            grad_g[idx] = self.gradient_scalar_sparse(
+                X_data, X_indptr, X_indices, y, w, Xw, j)
+
+        return grad_g
+
+    def gradient_scalar_sparse(self, X_data, X_indptr, X_indices, y, w, Xw, j):
+        grad_j = 0.
+        for i in range(X_indptr[j], X_indptr[j+1]):
+            grad_j += X_data[i] * (Xw[X_indices[i]] - y[X_indices[i]])
```

Comment:
A small remark,

since ``X_data[i] *  y[X_indices[i]]`` is a constant repeatedly recomputed, we could integrate it in ``initialize`` method, we sacrifice an overhead in the init to gain some computation later

though, I’m ok with not doing it now

Reply:
same with computing Xty in Lasso. I agree with you, let's open an issue but leave it for after this PR, which is already large